### PR TITLE
docs: readme

### LINKED ISSUE #16: Create a better README.md

### PR DESCRIPTION
**Pull request by danielorihuela**
## Description
Improve the README. Close #16.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by danielorihuela** [comment]
@rasswanth-s and I have started working on a better readme. We encourage people to give feedback and help us improve it.

**Comment by jmaunon** [comment]
> @rasswanth-s and I have started working on a better readme. We encourage people to give feedback and help us improve it.
> 
> **Note**: I started it in markdown mode to make my life easier. However, @jmaunon recomended to use rst to later on use it in the Sphinx documentation.

Hi @danielorihuela . If you want to use markdown is fine (it has no so many options as .rst but I agree that for most of the situations is enough). The only point is that a plugint would be needed (see for example: https://pypi.org/project/m2r2/)

File: README.md [hunk 1]
```diff
@@ -1,16 +1,37 @@
...
+Contributing
+------------
+We are open to colaboration! If you want to start contributing you only need to:
 
-## Disclaimer
+1. Check the [contributing guidelines](https://github.com/OpenMined/SyMPC/blob/main/CONTRIBUTING.md).
+2. Search for an issue in which you would like to work. Issues for newcomers are labeled with `good first issue`.
+3. Create a PR solving the issue.
 
-This library should not be used in a production environment because it is still a prototype.
```
Comment (gmuraru) [inline README.md:16]: Could you also leave this?

**Comment by gmuraru** [comment]
@danielorihuela is this still WIP?

**Comment by danielorihuela** [comment]
Yes, now that I ended the other. I am gonna try and improve this.

**Comment by danielorihuela** [comment]
@jmaunon I added some sections more. I do not know if would make sense to add even more sections.
I also added the rest version of the markdown readme. I do no have clear if I should let both readme or just keep the rest version.

**Comment by jmaunon** [comment]
@danielorihuela . Just keep one, it you already have in .rst I would keep this one because is easier to integrate with sphinx

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    <a href="http://duet.openmined.org/"><img src="https://github.com/OpenMined/design-assets/raw/master/logos/OM/mark-primary-trans.png" alt="SyMPC" width="200"></a>
+    <br>
+    SyMPC
+    <br>
+    </h1>
+    <h3 align="center">
+    <br>
+    A library that extends PySyft with SMPC support
+    <br>
+    </h3>
```
Comment (Syzygianinfern0) [inline README.rst:14]:
```suggestion
    SyMPC
    </h1>
    <h4 align="center">
    A library that extends PySyft with SMPC support
    <br>
    <br>
    </h4>
```

| Before | After |
| --- | --- |
| image | image |

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    <a href="https://codecov.io/gh/OpenMined/SyMPC"><img src="https://codecov.io/gh/OpenMined/SyMPC/branch/main/graph/badge.svg?token=TS2rZyJRlo" /></a>
+
+
+SyMPC **/ˈsɪmpəθi/** is a library which extends `PySyft <https://github.com/OpenMined/PySyft>`_ ≥0.3 with SMPC support. It allows to compute over encrypted data, to evaluate and to train neural networks.
+
+
+Installation
+############
+
+SyMPC is a companion library for PySyft. Therefore, we will need to install pysyft among other dependencies. We recommend using a virtual environment like conda.
```
Comment (Syzygianinfern0) [inline README.rst:27]:
```suggestion
SyMPC is a companion library for PySyft. Therefore, we will need to install PySyft among other dependencies. We recommend using a virtual environment like `conda`.
```

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    A library that extends PySyft with SMPC support
+    <br>
+    </h3>
+
+    <a href=""><img src="https://github.com/OpenMined/SyMPC/actions/workflows/tests.yml/badge.svg" /></a>
+    <a href="https://openmined.slack.com/messages/support"><img src="https://img.shields.io/badge/chat-on%20slack-7A5979.svg" /></a>
+    <a href="https://codecov.io/gh/OpenMined/SyMPC"><img src="https://codecov.io/gh/OpenMined/SyMPC/branch/main/graph/badge.svg?token=TS2rZyJRlo" /></a>
+
+
+SyMPC **/ˈsɪmpəθi/** is a library which extends `PySyft <https://github.com/OpenMined/PySyft>`_ ≥0.3 with SMPC support. It allows to compute over encrypted data, to evaluate and to train neural networks.
```
Comment (Syzygianinfern0) [inline README.rst:21]:
```suggestion
SyMPC **/ˈsɪmpəθi/** is a library which extends `PySyft <https://github.com/OpenMined/PySyft>`_ ≥0.3 with SMPC support. It allows computing over encrypted data, and to train and evaluate neural networks.
```

**Review by Syzygianinfern0 — CHANGES_REQUESTED** [review]
Minor changes to look, wording

File: README.rst (hunk 4 above)
Comment (danielorihuela) [reply README.rst]: Thanks for the feedback @Syzygianinfern0 !

**Review by Syzygianinfern0 — APPROVED** [review]

**Comment by danielorihuela** [comment]
> Sorry for the late review @danielorihuela.
> This looks great. One small addition you could make is to add instructions to run with docker.
> 
> Building
> `docker build -t sympc -f docker-images/Dockerfile . `
> 
> Running
> `docker run -i -t sympc`

Do not need to apologize! Check the docker section, see if you like it.

**Comment by jmaunon** [comment]
@kamathhrishi @danielorihuela , I have reviewed the docker section and it seems confusing. If you run those commands...later what? I have to enter into the docker container and... run the examples? (I do not think so because a Jupiter server is not running); start to develop? . It could be useful to include just a sentence with the next step you would expect from the person who reads that section

One more thing: `docker` section is actually a subsection of `Installation`, isn't it? (Actually is another way of installing sympc)

**Comment by danielorihuela** [comment]
@jmaunon 

> @kamathhrishi @danielorihuela , I have reviewed the docker section and it seems confusing. If you run those commands...later what? I have to enter into the docker container and... run the examples? (I do not think so because a Jupiter server is not running); start to develop? . It could be useful to include just a sentence with the next step you would expect from the person who reads that section
> 
> One more thing: `docker` section is actually a subsection of `Installation`, isn't it? (Actually is another way of installing sympc)

If I am not wrong, the docker image gives us another different isolated environment where we can do whatever we want. Let it be execute jupyter notebooks (if we use the correct options), run the tests, make our own experiments or modify the source code to contribute. I tried to make to reflect this ideas with a new structure following your recommendations @jmaunon. However, I am not sure if they are clearly enough or not. Also, if @kamathhrishi could check that I did not write anything technically wrong, that would be great.

**Comment by kamathhrishi** [comment]
@danielorihuela  You are right about the docker description. Just that I think the description could be worded like this:

_"If we cannot or do not want to use a virtual environment, we can use a docker image."_ (Don't use this, because people run docker for various reasons)

```
[keep the install docker part]

You can also run SyMPC using docker by running the following commands. 

.. code:: bash       
 $ docker build -t sympc -f docker-images/Dockerfile .   #builds image named sympc
$ docker run -i -t sympc #runs the container 
```

 Sorry I accidentally edited your comment by mistake.

File: README.rst [hunk 5]
```diff
@@ -0,0 +1,92 @@
+.. raw:: html
+
+    <h1 align="center">
+    <br>
+    <a href="http://duet.openmined.org/"><img src="https://github.com/OpenMined/design-assets/raw/master/logos/OM/mark-primary-trans.png" alt="SyMPC" width="200"></a>
+    <br>
+    SyMPC
+    </h1>
+    <h4 align="center">
```
Comment (rasswanth-s) [inline README.rst:9]:
```suggestion
    <h2 align="center">
```

File: README.rst [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    A library that extends PySyft with SMPC support
+    <br>
+    <br>
+    </h4>
```
Comment (rasswanth-s) [inline README.rst:13]:
```suggestion
    </h2>
```

**Review by rasswanth-s — APPROVED** [review]
Great work :+1:

**Comment by kamathhrishi** [comment]
> @kamathhrishi @danielorihuela , I have reviewed the docker section and it seems confusing. If you run those commands...later what? I have to enter into the docker container and... run the examples? (I do not think so because a Jupiter server is not running); start to develop? . It could be useful to include just a sentence with the next step you would expect from the person who reads that section
> 
> One more thing: `docker` section is actually a subsection of `Installation`, isn't it? (Actually is another way of installing sympc)

@jmaunon Sorry I didn't see your comment. Yeah, run examples or write and run code. Just that it helps have an environment that would surely work irrespective of platform. Its kind of under installation section. 

@danielorihuela You can place it under installation section with the description I mentioned above.

**Comment by danielorihuela** [comment]
> > @kamathhrishi @danielorihuela , I have reviewed the docker section and it seems confusing. If you run those commands...later what? I have to enter into the docker container and... run the examples? (I do not think so because a Jupiter server is not running); start to develop? . It could be useful to include just a sentence with the next step you would expect from the person who reads that section
> > One more thing: `docker` section is actually a subsection of `Installation`, isn't it? (Actually is another way of installing sympc)
> 
> @jmaunon Sorry I didn't see your comment. Yeah, run examples or write and run code. Just that it helps have an environment that would surely work irrespective of platform. Its kind of under installation section.
> 
> @danielorihuela You can place it under installation section with the description I mentioned above.

I have made a change. But I do not know if that's what you wanted @kamathhrishi.

**Comment by kamathhrishi** [comment]
@danielorihuela Looks good. Thanks for this PR :)

**Review by kamathhrishi — APPROVED** [review]

**Comment by kamathhrishi** [comment]
@gmuraru Want to have a look before merging?

**Pull request merged**